File: exemplo.c
#include "stdio.h"


/** 
 * Calcula a média dos números x e y.
 */
double media(double x, double y) {
    return (x + y) * 0.5;
}


int main() {
    double x, y;

    printf("x: ");
    scanf("%lf", &x);
    printf("y: ");
    scanf("%lf", &y);
    
    printf("Resultado: %lf\n", media(x, y));
}

File: exemplo.py
def media(*args, geometrica=False):
    """
    Calcula a média dos números x e y.
    """ 
    if geometrica:
        return prod(args) ** (1 / len(args))
    else:
        return sum(args) / len(args)


def prod(xs):
    """
    Calcula o produto da sequências de xs
    """
    prod = 1.0
    for x in xs:
        prod *= x
    return prod


def main():
    nums = []
    while (x := input(f'x{len(nums)}: ')):
        nums.append(float(x))
    
    print("Resultado: %s" % media(*nums))


if __name__ == '__main__':
    main()
